generated/src/aws-cpp-sdk-s3-crt/source/S3CrtErrorMarshaller.cpp

```diff
@@ -96,4 +101,43 @@ Aws::String S3CrtErrorMarshaller::ExtractEndpoint(const AWSError<CoreErrors>& er
   }
 
   return {};
+}
+AWSError<Aws::Client::CoreErrors> S3CrtErrorMarshaller::Marshall(const Aws::Http::HttpResponse& httpResponse) const
+{
+  //if response code not ok, use error marshaller
+  if(httpResponse.GetResponseCode() != HttpResponseCode::OK)
+  {
+    return XmlErrorMarshaller::Marshall(httpResponse);
+  }
+
+  Aws::String message{"Error in body of the response"};
+  //extract error message and code in the body
+  auto& body = httpResponse.GetResponseBody();
+  auto readPointer = body.tellg();
```

Comment:
check if the stream is valid, i.e. [`good`](https://en.cppreference.com/w/cpp/io/basic_ios/good) before doing operations on it.